src/azul/drs.py

```diff
@@ -123,13 +123,13 @@ def _get_object(self, drs_uri: str, access_method: AccessMethod) -> Access:
                 response = json.loads(response.data)
                 access_methods = response['access_methods']
                 method = one(m for m in access_methods if m['type'] == access_method.scheme)
-                access_id = method.get('access_id')
-                if access_id is None:
-                    access_url = method['access_url']
-                    require(access_url is not None)
+                access_url = method.get('access_url')
+                if access_url:
                     return Access(method=access_method,
                                   url=access_url['url'])
                 else:
+                    access_id = method['access_id']
+                    require(access_id is not None)
```

Comment:
This doesn't accurately address https://github.com/databiosphere/azul/issues/3226#issuecomment-878516143

> Tighten requirements around parsing the DRS URL

The main issue is that the caller (in the reproduction) continues to parse the URL without even checking that it is a `gs://` URL.